### PR TITLE
removed excess receiver protocols from enum

### PR DESCRIPTION
**Pull request by PietroPasotti**
## Issue
This helps address issue #https://github.com/canonical/tempo-coordinator-k8s-operator/issues/8

### CODE REVIEW

File: lib/charms/tempo_k8s/v2/tracing.py [hunk 1]
```diff
@@ -118,10 +118,6 @@ def __init__(self, *args):
 
 ReceiverProtocol = Literal[
     "zipkin",
-    "kafka",
-    "opencensus",
-    "tempo_http",
-    "tempo_grpc",
     "otlp_grpc",
```
Comment (mmkay) [inline lib/charms/tempo_k8s/v2/tracing.py:121]: The protocols aren't needed in `receiver_protocol_to_transport_protocol` either:

https://github.com/canonical/tempo-k8s-operator/pull/149/files#diff-b47b4334954f653c5e1d7a8831211b30280d18fa54a07681f9108cec9ce34923R140 

(where I somehow can't leave a comment)

Comment (PietroPasotti) [reply lib/charms/tempo_k8s/v2/tracing.py:121]: ehm, funny enough, it seems that Tempo is supporting jaeger thrift

Comment (PietroPasotti) [reply lib/charms/tempo_k8s/v2/tracing.py:121]: can go either way with it now. Do we add it to the supported protocols or strip it completely?

Comment (PietroPasotti) [reply lib/charms/tempo_k8s/v2/tracing.py:121]: ah but wait, we don't really care. This is monolithic tempo. IIRC we've removed it already in the distributed tempo?

Comment (PietroPasotti) [reply lib/charms/tempo_k8s/v2/tracing.py:121]: nope, still there.

Comment (mmkay) [reply lib/charms/tempo_k8s/v2/tracing.py:121]: It's in an uncertain state at the moment: https://github.com/canonical/tempo-coordinator-k8s-operator/blob/main/src/tempo.py#L53 still has it. I can remove it together with nginx PR if you think it's the way to go. 
@michaeldmitry, any opinions here maybe?

Comment (michaeldmitry) [reply lib/charms/tempo_k8s/v2/tracing.py:121]: My understanding is that we're going to drop Jaeger support at some point? https://www.jaegertracing.io/docs/1.58/apis/#opentelemetry-protocol-stable
That's why I didn't expose it as a config option, but looks like I missed it in https://github.com/canonical/tempo-coordinator-k8s-operator/blob/main/src/tempo.py#L53.

**Review by mmkay — APPROVED** [review]

**Comment by PietroPasotti** [comment]
maybe obsoleted by https://github.com/canonical/observability/pull/190

File: lib/charms/tempo_k8s/v2/tracing.py (hunk 1 above)
Comment (michaeldmitry) [reply lib/charms/tempo_k8s/v2/tracing.py:122]: Continuing in https://github.com/canonical/tempo-coordinator-k8s-operator/issues/8

**Comment by michaeldmitry** [comment]
Added supported protocols list as per ADR https://github.com/canonical/observability/pull/196

**Pull request merged**